src/LanguageServer/Test/GoToDefinitionTests.cs

```diff
@@ -81,10 +82,67 @@ class C:
             reference.range.Should().Be(17, 0, 17, 1); // TODO: store all locations
 
             reference = ds.FindDefinition(analysis, new SourceLocation(19, 5));
-            reference.range.Should().Be(5, 0, 9, 18);
+            reference.range.Should().Be(5, 6, 9, 18);
 
             reference = ds.FindDefinition(analysis, new SourceLocation(20, 5));
             reference.range.Should().Be(7, 4, 9, 18);
         }
+
+        [TestMethod, Priority(0)]
+        public async Task GotoModuleSource() {
+            const string code = @"
+import sys
+import logging
+
+logging.info('')
+";
+            var analysis = await GetAnalysisAsync(code, PythonVersions.LatestAvailable3X);
+            var ds = new DefinitionSource();
+
+            var reference = ds.FindDefinition(analysis, new SourceLocation(2, 9));
+            reference.Should().BeNull();
+
+            reference = ds.FindDefinition(analysis, new SourceLocation(5, 3));
+            reference.range.Should().Be(2, 7, 2, 14);
+
+            reference = ds.FindDefinition(analysis, new SourceLocation(3, 10));
+            reference.range.Should().Be(0, 0, 0, 0);
+            reference.uri.AbsolutePath.Should().Contain("logging");
+            reference.uri.AbsolutePath.Should().NotContain("pyi");
+
+            reference = ds.FindDefinition(analysis, new SourceLocation(5, 11));
+            reference.uri.AbsolutePath.Should().Contain("logging");
+            reference.uri.AbsolutePath.Should().NotContain("pyi");
+        }
+
+        [TestMethod, Priority(0)]
+        public async Task GotoModuleSourceImportAs() {
+            const string code = @"
+import logging as log
+log
+";
+            var analysis = await GetAnalysisAsync(code, PythonVersions.LatestAvailable3X);
+            var ds = new DefinitionSource();
+
+            var reference = ds.FindDefinition(analysis, new SourceLocation(3, 2));
+            reference.range.Should().Be(1, 18, 1, 21);
+
+            reference = ds.FindDefinition(analysis, new SourceLocation(2, 20));
+            reference.uri.AbsolutePath.Should().Contain("logging");
+            reference.uri.AbsolutePath.Should().NotContain("pyi");
+        }
+
+        [TestMethod, Priority(0)]
+        public async Task GotoBuiltinObject() {
+            const string code = @"
+class A(object):
+    pass
+";
+            var analysis = await GetAnalysisAsync(code);
+            var ds = new DefinitionSource();
+
+            var reference = ds.FindDefinition(analysis, new SourceLocation(2, 12));
+            reference.uri.AbsolutePath.Should().Contain("python.pyi");
```

Comment:
This file doesn't exist when I test the above. I get this error from VS Code:

```
Unable to open 'python.pyi': File not found (file:///c:/Users/jabaile/vscode-python/languageServer/903u0qhIUodb2IzkKD-YT8cai2jSRsVAAP-6wu9bc44=/python.pyi).
```

Reply:
Yes it is fake. Release build doesn't navigate to stubs/caches. Debug does it for convenience in case you want to find the stub easily and look there.

Reply:
This is on a release build produced by `dotnet build -c Release`.